a new set of patches from mame !

zelurker opened the pull request:
this time I had to pre-process the patches with a perl script, but quite some stuff must be done manually because of the growing differences...
I stopped after adding the 68030, + ec030 and ec040 (and scc68070 which was in the way, but is just a 68010 with a 32 bit data bus, so there is nothing specific to this one except the initialization).

I tried to add something in m68kconf.h, just a define for M68K_EMULATE_030, it should be enough normally.

Updated very shortly the doc to update the list of what's emulated, and updated the version number to follow what's in mame, it's 4.10 now.
I just tested by compiling it, I didn't try to run anything specific, but it should be ok to merge that normally, even if it's more error prone than the previous patches...

kstenerud commented on m68kcpu.c:
```diff
@@ -122,16 +127,16 @@ const uint m68ki_shift_32_table[65] =
 {
 	{ /* 000 */
-		  4, /*  0: Reset - Initial Stack Pointer                      */
+		 40, /*  0: Reset - Initial Stack Pointer                      */
```
Why is this one changed?

kstenerud commented on m68kcpu.c:
```diff
@@ -198,7 +203,7 @@ const uint8 m68ki_exception_cycle_table[4][256] =
 	},
 	{ /* 010 */
-		  4, /*  0: Reset - Initial Stack Pointer                      */
+		 40, /*  0: Reset - Initial Stack Pointer                      */
```
Similar here

zelurker commented:
Well it's related to the commit "CPUs actually take some time to reset.
Changed the 68000/68010 to eat an appropriate number of cycles after a
reset.".
Now don't ask me why it's coded like that, I am at a loss, I thought there
was already CYC_RESET for that, but I guess one is related to the reset
instruction and the other to the exception itself. Well apparently the
exact number of cycles for the exception was apparently found, so they
changed it here, and I merged the patch !


Le jeu. 12 déc. 2019 à 09:09, Karl Stenerud <notifications@github.com> a
écrit :

> *@kstenerud* requested changes on this pull request.
>
> Wow! This is super impressive! Thanks so much!!
>
> The only thing left is to double-check the two timings.
> ------------------------------
>
> In m68kcpu.c
> <https://github.com/kstenerud/Musashi/pull/50#discussion_r357002766>:
>
> >  {
>  	{ /* 000 */
> -		  4, /*  0: Reset - Initial Stack Pointer                      */
> +		 40, /*  0: Reset - Initial Stack Pointer                      */
>
> Why is this one changed?
> ------------------------------
>
> In m68kcpu.c
> <https://github.com/kstenerud/Musashi/pull/50#discussion_r357002832>:
>
> > @@ -198,7 +203,7 @@ const uint8 m68ki_exception_cycle_table[4][256] =
>  		  4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4
>  	},
>  	{ /* 010 */
> -		  4, /*  0: Reset - Initial Stack Pointer                      */
> +		 40, /*  0: Reset - Initial Stack Pointer                      */
>
> Similar here
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kstenerud/Musashi/pull/50?email_source=notifications&email_token=AAV3SD3FRN3P6KMYD25HK7DQYHWRZA5CNFSM4JZWSAD2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCO5WCFA#pullrequestreview-331047188>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAV3SD5R5FRLLW3MFIH5JXLQYHWRZANCNFSM4JZWSADQ>
> .
>

kstenerud commented:
Hmm ok. Strange that they only did it for two of the CPU types... But we can always sort that out later. Merging!

zelurker commented:
Also on 2nd thought I should have put the reset_cycles variable inside the
cpu context instead of this weird global m68ki_reset_cycles. It works, but
you are not supposed to change the cpu context between a reset and a
function to execute cycles... I can add a commit to replace this by a a
variable in the cpu context if you want (this was one of the 1st patches
and I was not sure how to handle it).

Le jeu. 12 déc. 2019 à 09:43, Emmanuel Anne <emmanuel.anne@gmail.com> a
écrit :

> Well it's related to the commit "CPUs actually take some time to reset.
> Changed the 68000/68010 to eat an appropriate number of cycles after a
> reset.".
> Now don't ask me why it's coded like that, I am at a loss, I thought there
> was already CYC_RESET for that, but I guess one is related to the reset
> instruction and the other to the exception itself. Well apparently the
> exact number of cycles for the exception was apparently found, so they
> changed it here, and I merged the patch !
>
>
> Le jeu. 12 déc. 2019 à 09:09, Karl Stenerud <notifications@github.com> a
> écrit :
>
>> *@kstenerud* requested changes on this pull request.
>>
>> Wow! This is super impressive! Thanks so much!!
>>
>> The only thing left is to double-check the two timings.
>> ------------------------------
>>
>> In m68kcpu.c
>> <https://github.com/kstenerud/Musashi/pull/50#discussion_r357002766>:
>>
>> >  {
>>  	{ /* 000 */
>> -		  4, /*  0: Reset - Initial Stack Pointer                      */
>> +		 40, /*  0: Reset - Initial Stack Pointer                      */
>>
>> Why is this one changed?
>> ------------------------------
>>
>> In m68kcpu.c
>> <https://github.com/kstenerud/Musashi/pull/50#discussion_r357002832>:
>>
>> > @@ -198,7 +203,7 @@ const uint8 m68ki_exception_cycle_table[4][256] =
>>  		  4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4
>>  	},
>>  	{ /* 010 */
>> -		  4, /*  0: Reset - Initial Stack Pointer                      */
>> +		 40, /*  0: Reset - Initial Stack Pointer                      */
>>
>> Similar here
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/kstenerud/Musashi/pull/50?email_source=notifications&email_token=AAV3SD3FRN3P6KMYD25HK7DQYHWRZA5CNFSM4JZWSAD2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCO5WCFA#pullrequestreview-331047188>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAV3SD5R5FRLLW3MFIH5JXLQYHWRZANCNFSM4JZWSADQ>
>> .
>>
>

kstenerud commented:
Hmm yeah that actually sounds cleaner tbh.
